### DISCUSSION TITLE
Struggling with Aquarite - maybe I just don't know what to do?

### DISCUSSION THREAD

**Discussion by skipfire** [tagyoureit/nodejs-poolController#439] [App and O/S Config Questions] [answered]
I am trying to get an Aquarite salt system recognized, and this is my first foray with nodejs-poolController so I quite possibly have no clue what I'm doing with the software.  I am using a Raspberry Pi 3B and Waveshare Industrial USB RS-485 adapter for the hardware.  I did find the chlorinator wiki page and issue #293.  The Pi does recognize the RS-485 adapter and I have added 12v power to the RS-485 pins on the Aquarite.  I have connected the wires as indicated with green to ground, yellow to B-, black to A+.  I have verified with a multi-meter that I have 12v at the Aquarite and I see a bit over 2v on the D+ and D-, without the 12v connected both were at 0v so I believe I am getting it powered up.

I have tried deleting the 2 data files and the config.json and then running npm start, however nothing I do seems to get me away from `Inactivity timeout for 0 serial port /dev/ttyUSB0 after 10 seconds` showing up every 10 seconds.  In the poolConfig.json the chlorinators is just empty array.

I'm hoping someone can shed some light on what I'm doing wrong.  I'm not sure what to do with the commands information in the chlorinator wiki, but I'm guessing my problem is before that right now.

**Comment by rstrouse** [comment, marked as answer]
First, some models of the AquaRite require 10v on the power pins for the RS485 connection.  Then if you did not define the chlorinator as a Nixie controlled item it will not send any messages to the chlorinator.  If this is the only thing on the bus then there will be no traffic and njsPC will reset the connection just to make sure there is not a comms error with the adapter.

In dashPanel under the gears icon in the upper right click on the Chemistry tab.  Then click Add Controller.  Select Nixie from the chlorinator dropdown the press the chlorinator button.  Add in the attributes for the chlorinator then press the Save Chlorinator button.

After njsPC is aware that an AquaRite chlorinator exists it will start attempting to contact it.

**Comment by skipfire** [comment]
@rstrouse thanks, I am supplying power to the AquaRite on the RS485 connection.  Based on what you have responded with I obviously misunderstood the readme about the clients, I understood it to mean there was no UI but I thought stuff auto-detected from some of the statements.  From your post I now understand I need to install a client to be able to do manual configuration to move to the next steps.  I'll do that this evening and hopefully post back success in the next few hours.

**Reply by rstrouse** [reply]
Install the dashPanel webserver.  It will allow you to configure your installation and troubleshoot any issues you may have.  I'll have a look at the readme file.

**Reply by skipfire** [reply]
Thanks for the help, I got DashPanel and it took a lot of guess work to get it, but I saw the messages stop and I have a salt level showing that matches the display.

Now I just need to figure out if I can get more data than the default shows, but since it is working I may be able to figure it out, or contribute if it isn't supported yet.  I was hoping for all the diagnostic values to be visible, or at least the "No Flow" signal so that I can setup alerts when the switch breaks again.

**Reply by rstrouse** [reply]
Those will show in the status line.  The following status values are detected and included in the salt level status message.  If no flow is detected it will show under the chlorinator line.
```
    [0, { name: 'ok', desc: 'Ok' }],
    [1, { name: 'lowflow', desc: 'Low Flow' }],
    [2, { name: 'lowsalt', desc: 'Low Salt' }],
    [3, { name: 'verylowsalt', desc: 'Very Low Salt' }],
    [4, { name: 'highcurrent', desc: 'High Current' }],
    [5, { name: 'clean', desc: 'Clean Cell' }],
    [6, { name: 'lowvoltage', desc: 'Low Voltage' }],
    [7, { name: 'lowtemp', desc: 'Water Temp Low' }],
    [8, { name: 'commlost', desc: 'Communication Lost' }]
```

**Reply by skipfire** [reply]
For some reason it isn't showing data that matches the LEDs on the controller `"status":{"val":0,"name":"ok","desc":"Ok"}` when the low-flow LED is lit, but I do at least see the data.  I'll work with it and see if I can figure out the last, maybe I have the wrong model number selected or something.

**Reply by rstrouse** [reply]
If you load up messageManager you can see the comms.  It may not be reporting.  You already have it installed so if you type http://[address of dashPanel]:5150/messageManager.html you can see all the traffic.

**Reply by skipfire** [reply]
Nothing shows in messageManager.html, I've tried turning the log on and off multiple times with the button in the top right, also tried turning on superchlorinate and the physical panel did not indicate that either (the software does show the timer as does `:4200/state/all`), the lastComm value does increment as I would expect and does equate to now but that's the only sign that it is communicating. Is there a log file with the messages?

**Comment by rstrouse** [comment]
Here is a wiki to get you started.
https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager

**Reply by skipfire** [reply]
Thanks, I have the message manager working now, though the data coming from the panel on the salt response always has a value of 78,0; so it seems to be bad data from the panel?  The 78 works out to 3900ppm which matches the display on the panel and I've seen that value change multiple times in the data coming back today.

**Comment by rstrouse** [comment]
An example of a message that contains very low salt.  Bit 3 is set on the status value.  Bit 8 is ignored as this is a bit that indicates communication lost when the chlorinator is connected to an intermediate device like IntelliChem.  When that equipment is not connected then bit 8 has no meaning.
`[16,2,0,18][52,132][220,16,3]`

Index | Value | Description
-- | -- | --
0 | 52 | The current salt level   for the chlorinator / 50.  Salt level =   byte(0) * 50 = 52 * 50 = 2600
1 | 132 | Status flags bitmask for   the chlorinator.   0=Ok   1=Low Flow   2=Low Salt   3=Very Low Salt   4=High Current   5=Clean Cell   6=Low Voltage   7=Low Temp   8=Empty bit.

**Reply by skipfire** [reply]
That looks like the except from the chlorinator wiki page here that I was referencing, and the messages are 76, 0 even with the panel showing low-flow where I would expect 76, 1.  This image was taken from the log just a minute ago and the pump had been off for about 90 minutes at this point.

![image](https://user-images.githubusercontent.com/10307327/167968832-d58cc4a4-e486-41ad-85c3-314821ac6647.png)

**Reply by rstrouse** [reply]
Which panel is showing low flow.  Unfortunately, this is the only message returned over RS485.  Not even all revisions of IntelliChlor report all the values.  Does it report 128 when the flow is good and there are no other alerts?

**Reply by skipfire** [reply]
The physical panel is what shows the low-flow.  When the pump is running it also shows 76, 0.

![20220511_194344](https://user-images.githubusercontent.com/10307327/167969967-10afcfc9-da33-4e1d-8679-15d4b47d96aa.jpg)

**Comment by tagyoureit** [comment]
Per the manual, this control box needs to be installed with a separate flow switch.  Do you have a flow switch outside of the salt cell?
<img width="1381" alt="2022-05-12_09-31-58" src="https://user-images.githubusercontent.com/1646840/168124478-46a8b4f1-980f-42e5-acbb-398ee815b7d5.png">

**Comment by skipfire** [comment]
@tagyoureit yes I do have the flow sensor installed, and the physical panel displays the correct low flow state at all times (lit when pump is off, unlit when pump is running).  That switch being faulty was my main motivation for installing this, so that I could compare pump state to the salt system's low-flow and be alerted when the switch fails.  When the flow sensor is not connected or failed the salt system does not generate, so it isn't an optional component.

**Reply by tagyoureit** [reply]
Just a theory, but I think that the chlor may only generate the low flow warning when it is being asked to generate chlorine but there is no flow.  In other words, the external flow switch may act as a kill switch and hence the chlor knows it shouldn't be generating chlorine.  A test might be to force the flow switch to think there is flow (close the circuit) and then ask the chlor to generate chlorine and see if it throws an error in the rs485 messages.

**Reply by skipfire** [reply]
My current low-flow switch is working fine and the test you describe is the pump running state. When the pool pump is running the generating light is lit, and the low-flow is dark. When the pump is not running the low-flow lights up and the generating light goes dark as the flow sensor is a cutoff for the cell so that it does not damage it by trying to generate chlorine when the water is not flowing. The SWG and pump have no interface with each other than the pool pump pushing enough water to close the low-flow switch to enable the system. Physically everything is working (I replaced the low-flow switch last week), the RS485 data just doesn't match the physical low-flow indicator light.

**Reply by tagyoureit** [reply]
Ah, that's it!  The Intellichlor has an internal flow switch and so the salt cell itself knows about the flow sensor.  I was thinking there was a 2nd internal flow switch in that cell but that isn't the case.  With the Aquarite, the flow switch is connected directly to the control box.  The salt cell knows nothing about the flow of the water.  It is the control box that determines when to power the cell and for how long.  The cell only responds back with what it knows - the salinity (and maybe current output?? not sure).  You will not be able to get the flow sensor status from the RS485 packets because they are coming from the salt cell.  If you do want to read that sensor you would need to attach it to a dry trigger input (in addition to the control box) and monitor the status that way.  REM can do that for you with a compatible input board like the Sequent Microsystems boards.

**Reply by skipfire** [reply]
Thanks, it's a bit of a bummer they built it that way since the wires physically connect to the control box, but what you said made sense.
